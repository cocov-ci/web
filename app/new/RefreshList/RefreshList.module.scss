@import "styles/variables.module";

.refreshList {
  display: flex;
  align-items: center;
  padding: 14px 30px;
  border-bottom: 1px solid $colorGray;

  .description {
    flex: 1;
    margin-left: 10px;
  }
}
