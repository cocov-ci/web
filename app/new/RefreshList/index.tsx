'use client'

import { Clock } from 'lucide-react'

import Button from 'app/common/Button'
import RelativeTime from 'app/common/RelativeTime'
import Text from 'app/common/Text'

import styles from './RefreshList.module.scss'

interface RefreshListParams {
  date: string
  onRefresh: () => void
  loading: boolean
}

const RefreshList = ({ date, onRefresh, loading }: RefreshListParams) => {
  return (
    <div className={styles.refreshList}>
      <Clock size={19} />
      <Text className={styles.description}>
        This list was last updated <RelativeTime timestamp={new Date(date)} />.
        You can refresh it in case you feel like something is missing.
      </Text>
      <Button disabled={loading} onClick={() => onRefresh()} style="mini">
        Refresh now
      </Button>
    </div>
  )
}

export default RefreshList
