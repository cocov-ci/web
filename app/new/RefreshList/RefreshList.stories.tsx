import type { Meta, StoryObj } from '@storybook/react'

import RefreshList from './index'

const meta: Meta<typeof RefreshList> = {
  title: 'Repositories/New/RefreshList',
  component: RefreshList,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof RefreshList>

export const Default: Story = {
  args: {
    loading: false,
    date: '2022-11-22T21:06:17.000Z',
    onRefresh: () => null,
  },
}
