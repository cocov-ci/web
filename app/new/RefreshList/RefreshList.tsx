import type { Meta, StoryObj } from '@storybook/react'

import RefreshList from './index'

const meta: Meta<typeof RefreshList> = {
  title: 'Repositories/New/RefreshList',
  component: RefreshList,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof RefreshList>

export const Default: Story = {
  args: {},
}
