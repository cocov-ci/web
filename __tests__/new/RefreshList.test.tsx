import { fireEvent, render, screen } from '@testing-library/react'

import RefreshList from 'app/new/RefreshList'

import '@testing-library/jest-dom'

const args = {
  loading: false,
  date: new Date(new Date().setDate(new Date().getDate() - 1)).toDateString(),
  onRefresh: () => null,
}

describe('new/RefreshList', () => {
  it('renders RefreshList', () => {
    const { container } = render(<RefreshList {...args} />)

    expect(container.querySelector('.description')?.textContent).toEqual(
      'This list was last updated 2 days ago. You can refresh it in case you feel like something is missing.',
    )
  })

  it('renders RefreshList with "loading" param', () => {
    const { container } = render(<RefreshList {...args} loading={true} />)

    expect(container.querySelector('button')).toHaveAttribute('disabled')
  })

  it('triggers "onRefresh" event', () => {
    const handleEvent = jest.fn()
    render(<RefreshList {...args} onRefresh={handleEvent} />)

    fireEvent.click(screen.getByRole('button'))
    expect(handleEvent).toHaveBeenCalledTimes(1)
  })

  it('renders RefreshList snapshots', () => {
    const { container } = render(<RefreshList {...args} />)

    expect(container).toMatchSnapshot()
  })
})
