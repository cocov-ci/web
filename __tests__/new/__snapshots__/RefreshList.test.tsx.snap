// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/RefreshList renders RefreshList snapshots 1`] = `
<div>
  <div
    class="refreshList"
  >
    <svg
      class="lucide lucide-clock"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="12 6 12 12 16 14"
      />
    </svg>
    <p
      class="text description"
    >
      This list was last updated 
      <span
        class=""
        title="Mar 30, 2023, 12:00 AM GMT+1"
      >
        2 days ago
      </span>
      . You can refresh it in case you feel like something is missing.
    </p>
    <button
      class="className button mini"
    >
      Refresh now
    </button>
  </div>
</div>
`;
